### PR TITLE
Add Opcodes and Errors classes in generated Typescript code

### LINKED ISSUE #1033: Enum "Opcodes" and "Errors" are possibly missing in generated .ts file
When writing sandbox test it is very convenient to check if transaction with exact op was completed (or aborted with exact exit_code). Adding this will definitely make writing tests more convenient.
Some example of possible syntax:
```
export abstract class Op {
    static transfer = 0xf8a7ea5;
    static transfer_notification = 0x7362d09c;
    static internal_transfer = 0x178d4519;
    static excesses = 0xd53276db;
    static burn = 0x595f07bc;
    static burn_notification = 0x7bdd97de;
    
    static provide_wallet_address = 0x2c76b973;
    static take_wallet_address = 0xd1735400;
    static mint = 21;
    static change_admin = 3;
    static change_content = 4;
}

export abstract class Errors {
    static invalid_op = 709;
    static not_admin  = 73;
    static unouthorized_burn = 74;
    static discovery_fee_not_matched = 75;
    static wrong_op = 0xffff;
    static not_owner = 705;
    static not_enough_ton = 709;
    static not_enough_gas = 707;
    static not_valid_wallet = 707;
    static wrong_workchain = 333;
    static balance_error   = 706;
}
```

### PR DESCRIPTION
**Pull request by Shvandre**
## Issue

Closes #1033 .

## Checklist

- [x] I have updated CHANGELOG.md
- [ ] I have documented my contribution in `docs/` and made the build locally
- [ ] I have added tests to demonstrate the contribution is correctly implemented: this usually includes both positive and negative tests, showing the happy path(s) and featuring intentionally broken cases
- [x] I have run all the tests locally and no test failure was reported
- [x] I have run the linter, formatter and spellchecker
- [x] I did not do unrelated and/or undiscussed refactorings

### CODE REVIEW

**Review by anton-trunov — CHANGES_REQUESTED** [review]
let's either
- add tests with the new functionality, or
- simplify the existing tests using it

**Comment by anton-trunov** [comment]
image

apparently this is not entirely true :)
CI is red because Prettier is unhappy

File: src/bindings/writeTypescript.ts [hunk 1]
```diff
@@ -217,6 +231,23 @@ export function writeTypescript(
         }
     });
     w.append(`]`);
+
+    //Opcodes
+    //So user can use them in sandbox tests
+    w.append("export abstract class Opcodes {");
```
Comment (verytactical) [inline src/bindings/writeTypescript.ts:237]: TypeScript has `enum` syntax for this. `abstract class` implies user should inherit from it to get access to its fields, and that's definitely not the way in a language without multiple inheritance.

File: src/bindings/writeTypescript.ts [hunk 2]
```diff
@@ -272,58 +303,66 @@ export function writeTypescript(
 
         if (init) {
             w.append(
-                `static async init(${writeArguments(init.args).join(", ")}) {`,
+                `static async init<T extends ${abi.name}>(` +
+                    `this: new (address: Address, init?: { code: Cell, data: Cell }) => T,` +
```
Comment (verytactical) [inline src/bindings/writeTypescript.ts:307]: This kind of types is meant to type legacy JS code, not to write or generate novel code.

File: src/bindings/writeTypescript.ts [hunk 3]
```diff
@@ -192,6 +192,20 @@ export function writeTypescript(
     }
 
     // Errors
+    // Map, so user can do something like ExitCodes["Stack Overflow"]
+    w.append(`const ${abi.name}_errorMessages: { [key: string]: number } = {`);
+    w.inIndent(() => {
+        if (abi.errors) {
+            Object.entries(abi.errors).forEach(([k, abiError]) => {
```
Comment (verytactical) [inline src/bindings/writeTypescript.ts:199]: `for (const [k, abiError] of abi.errors)` would be more readable, concise and performant.

File: src/bindings/writeTypescript.ts [hunk 4]
```diff
@@ -192,6 +192,20 @@ export function writeTypescript(
...
+                const escapedMessage = abiError.message.replaceAll('"', '\\"');
+                w.append(`    "${escapedMessage}": ${k},`);
```
Comment (verytactical) [inline src/bindings/writeTypescript.ts:201]: This is an indent that doesn't take `.inIndent()` state in consideration.

File: src/bindings/writeTypescript.ts (hunk 3 above)
Comment (anton-trunov) [reply src/bindings/writeTypescript.ts:199]: @verytactical there is a few of instances of `Object.entries(foo).forEach(([k, v]) ...` in the codebase. Could you please open an issue to refactor those?

**Comment by anton-trunov** [comment]
Let's move on with this PR please. @Shvandre can we see some tests for the changes?

**Comment by Shvandre** [comment]
Decided to put this into the backlog. Will return to this after Typescript codegen refactoring

**Pull request closed without merging**